### ISSUE TITLE
IrGraphGenerator is broken

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#129]
## 🐛 Bug

IrGraphGenerator fails to pass with the current develop branch.

```
[ RUN      ] JitTest.GPU_IrGraphGenerator_CUDA
unknown file: Failure
C++ exception with description "Cannot reduce an axis that is marked as broadcasted as it has an undetermined size. Tried to reduce ID = 0x5590b6d326e0 of tensor 0x5590bf5f0cb0
Exception raised from newForReduction at ../torch/csrc/jit/codegen/cuda/arith.cpp:394 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x68 (0x7f7f9c409d78 in /home/nmaruyama/pytorch/src/csarofeen.3/build/lib/libc10.so)
frame #1: torch::jit::fuser::reductionOp(torch::jit::fuser::BinaryOpType, std::vector<int, std::allocator<int> > const&, torch::jit::fuser::Val*, torch::jit::fuser::TensorView*) + 0x1705 (0x7f7f9fd70be5 in /home/nmaruyama/pytorch/src/csarofeen.3/build/lib/libtorch_cuda.so)
frame #2: torch::jit::testGPU_IrGraphGenerator() + 0x23a (0x559080494a0a in ./build/bin/test_jit)
```

This is likely because of an added check at #126.

**Comment by tlemo** [comment]
@csarofeen, this is the fusion IR used to test the IrGrapGenerator. If it's not valid we can change it (the goal is just to cover all the node types)

**Comment by naoyam** [comment]
Fixed in #138
